File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/waterfall/WaterfallPagerAdapter.java
package com.twtstudio.tjwhm.lostfound.waterfall;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by tjwhm on 2017/7/3.
 **/

public class WaterfallPagerAdapter extends FragmentPagerAdapter {

    private List<Fragment> fragmentsOfWaterfall = new ArrayList<>();
    private List<String> fragmentsTitles = new ArrayList<>();

    public WaterfallPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    public void add(Fragment fragment, String title){
        fragmentsOfWaterfall.add(fragment);
        fragmentsTitles.add(title);
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentsOfWaterfall.get(position);
    }

    @Override
    public int getCount() {
        return fragmentsOfWaterfall.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return fragmentsTitles.get(position);
    }
}

File: WePeiYang/lostfound2/src/main/java/com/yookiely/lostfond2/waterfall/WaterfallPagerAdapter.kt
package com.yookiely.lostfond2.waterfall

import android.support.v4.app.FragmentManager
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import java.util.ArrayList

class WaterfallPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val fragmentsOfWaterfall = ArrayList<Fragment>()
    private val fragmentsTitles = ArrayList<String>()

    fun add(fragment: Fragment, title: String) {
        fragmentsOfWaterfall.add(fragment)
        fragmentsTitles.add(title)
    }

    override fun getItem(position: Int): Fragment = fragmentsOfWaterfall[position]

    override fun getCount(): Int = fragmentsOfWaterfall.size

    override fun getPageTitle(position: Int): CharSequence? = fragmentsTitles[position]
}

